CandidateReranking.md

```diff
@@ -0,0 +1,368 @@
+| profiles | [Profile[]](#profile) | **Required**: Array of profiles to be reranked.
+
+### Query
+
+----------
+The query and its child objects are expected to be taken from the response of the [SemanticSearchAPI](https://github.com/careerbuilder/DataScienceAPIDocumentation/blob/master/SemanticSearchV2.md). All the following objects mirror that response.
+
+#### Query
+| Param    | Type | Description
+|----------|------|--------|
+| search_keywords | String | **Optional**: Raw query string applied in original search.
+| location | String | **Optional**: Location requested in search.
+| should_use_radius | boolean | **Optional**: Flag on the original search to determine use of radius.
+| radius | Integer | **Optional**: Radius used in original search.
+| extracted_keywords | [Keyword[]](#keyword) | **Optional**: Array of **Keyword** enrichments to the query string or `search_keywords` returned from SemanticSearchAPI.
+| parsed_input | ParsedInput | **Optional**: **ParsedInput** Object comprised of a Map<String,String> as `input_to_extracted_keywords` and a list of [ParsedEntities](#ParsedInput)
+| filters | [Filters[]](#Filter) | **Optional**: Array of filters used in the original query to search.
+
+#### ParsedInput
+| Param    | Type | Description
+|----------|------|--------|
+| input_to_extracted_keywords | Map<String, String> | **Optional**: Words extracted from `search_keywords` as the **Key** and their relavtive keywords identified by SemanticSearchAPI as the **Value**.
+| parsed_entities | ParsedEntity | **Optional**: Entities parsed from `search_keywords`.
+
+#### ParsedEntity
+| Param    | Type | Description
+|----------|------|--------|
+| entity_name | String | **Optional**: Name of the entity.
+| is_phrase | Boolean | **Optional**: Flags the entity as a phrase or not.
+| is_selected | Boolean | **Optional**: Flags the entity as sleceted or not.
+
+#### Keyword
+Entites returned from the SemanticSearchAPI.
+
+| Param    | Type | Description
+|----------|------|--------|
+| name | String | **Optional**: Name of the keyword, e.g. "Java".
+| type | String | **Optional**: Type of the keyword, e.g. "skill".
+| weight | Integer | **Optional**: Weight of the keyword as an int. \[0-100\]
+| relationships | [Relationship](#relationship) | **Optional**: Array of relationship objects which enrich the keyword with additional entities.
+
+#### Relationship
+A `relationship` is composed of a **\<semantic enrichment type\>** and the associated set of entities which can be classified under that **\<semantic enrichment type\>**. Accepted values for the **\<semantic enrichment type\>** include: `job_titles`, `occupations`, `skills`, `job_level`, `text_kernel_related_keywords`, `interesting_keywords`, `related_keywords`, `related_keywords_recruiter`, `related_keywords_jobseeker`, `raw_job_titles_jobseeker`, `raw_job_titles_recruiter`, `extracted_kewords`, `custom_keywords`, `related_search_terms`, and `job_titles_recruiter`.
```

Comment:
Is this an exhaustive list? If so then we should say "Accepted values for the **\<semantic enrichment type\>** are: ...."